hail_search/queries/mito.py

```diff
@@ -118,6 +122,178 @@ class MitoHailTableQuery(BaseHailTableQuery):
+            project_ht_dict[sample_type] = (ht, sample_data[sample_type])
+        all_project_hts.append(project_ht_dict)
+
+    def _filter_entries_ht_both_sample_types(
+        self, wes_ht, wes_project_samples, wgs_ht, wgs_project_samples, inheritance_filter=None, quality_filter=None,
+        is_merged_ht=False, **kwargs
+    ):
+        wes_ht, sorted_wes_family_sample_data = self._add_entry_sample_families(wes_ht, wes_project_samples, is_merged_ht)
+        wgs_ht, sorted_wgs_family_sample_data = self._add_entry_sample_families(wgs_ht, wgs_project_samples, is_merged_ht)
+        wes_ht = wes_ht.rename({'family_entries': SampleType.WES.family_entries_field})
+        wgs_ht = wgs_ht.rename({'family_entries': SampleType.WGS.family_entries_field})
+        ht = wes_ht.join(wgs_ht, how='outer')
+
+        sample_types = [
+            (SampleType.WES, sorted_wes_family_sample_data),
+            (SampleType.WGS, sorted_wgs_family_sample_data)
+        ]
+        for sample_type, _ in sample_types:
+            ht = self._filter_quality(
+                ht, quality_filter,
+                annotation=sample_type.passes_quality_field, entries_ht_field=sample_type.family_entries_field, **kwargs
+            )
+
+        ch_ht = None
+        for sample_type, family_sample_data in sample_types:
+            ht, ch_ht = self._filter_inheritance(
+                ht, ch_ht, inheritance_filter, family_sample_data,
+                annotation=sample_type.passes_inheritance_field, entries_ht_field=sample_type.family_entries_field
+            )
+
+        family_idx_map = self._build_family_index_map(sample_types, sorted_wes_family_sample_data, sorted_wgs_family_sample_data)
+        ht = self._apply_multi_sample_type_entry_filters(ht, family_idx_map)
+        ch_ht = self._apply_multi_sample_type_entry_filters(ch_ht, family_idx_map)
+
+        return ht, ch_ht
+
+    @staticmethod
+    def _build_family_index_map(sample_types, sorted_wes_family_sample_data, sorted_wgs_family_sample_data):
+        family_guid_idx_map = defaultdict(dict)
+        for sample_type, sorted_family_sample_data in sample_types:
+            for family_idx, samples in enumerate(sorted_family_sample_data):
+                family_guid = samples[0]['familyGuid']
+                family_guid_idx_map[family_guid][sample_type.value] = family_idx
+        return hl.dict(family_guid_idx_map)
+
+    def _apply_multi_sample_type_entry_filters(self, ht, family_idx_map):
+        if ht is None:
+            return ht
+
+        # Keep family from both sample types if either passes quality AND inheritance
+        for sample_type in SampleType:
+            ht = ht.annotate(**{
+                sample_type.family_entries_field: hl.enumerate(ht[sample_type.family_entries_field]).starmap(
+                    lambda i, samples: hl.or_missing(
+                        (hl.case()
+                        .when(hl.is_defined(samples[0]['familyGuid']),
```

Comment:
So a case statement with exactly one clause is unnecessary. Also, I think this would actually be more readable if you don't have the current helper function and instead move more of the logic inline here and just have a helper function for the shared code of evaluating whether a family passes for a given sample type. Then here use `hl.bind` to evaluate the other sample type index. Also, per my comment on the last review, we should handle the case where the first sample is missing a call but other samples in the family are not. You should be able to rewrite this as 
```
 hl.or_missing(
    hl.bind(
        lambda other_sample_type_idx: (
            self._family_has_valid_sample_type_entries(ht, sample_type, i) | 
            self._family_has_valid_sample_type_entries(ht, sample_type.other_sample_type, other_sample_type_idx)
        ),
        family_idx_map.get(samples.find(hl.is_defined['familyGuid']).get(sample_type.other_sample_type.value),
   ), samples)

...

@staticmethod
def _family_has_valid_sample_type_entries(ht, sample_type, sample_type_family_idx):
    return (
        hl.is_defined(sample_type_family_idx) &
        hl.is_defined(ht[sample_type.passes_quality_field][sample_type_family_idx]) &
        hl.is_defined(ht[sample_type.passes_inheritance_field][sample_type_family_idx])
    )
```

Reply:
This is much cleaner, thank you for helping me write it. I'm just tweaking `samples.find(hl.is_defined['familyGuid'])` to `hl.coalesce(samples)[0]['familyGuid']`.